src/test/java/domain/ladder/LadderStepTest.java

```diff
@@ -0,0 +1,57 @@
+package domain.ladder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import domain.generator.BooleanGenerator;
+import domain.generator.FalseBooleanGenerator;
+import domain.generator.TrueBooleanGenerator;
+import domain.ladder.LadderStep;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class LadderStepTest {
+
+    private final BooleanGenerator existsGenerator = new TrueBooleanGenerator();
+    private final BooleanGenerator noneGenerator = new FalseBooleanGenerator();
+
+    @DisplayName("첫 번째 사다리 발판을 생성할 때는 제약 없이 생성한다.")
+    @Test
+    void createFreelyTest() {
+        Assertions.assertThat(LadderStep.createFreely(existsGenerator)).isEqualTo(LadderStep.EXISTS);
+        assertThat(LadderStep.createFreely(noneGenerator)).isEqualTo(LadderStep.NONE);
+    }
+
+    @DisplayName("연속해서 사다리 발판이 존재할 수 없다.")
+    @Nested
+    class createConsideringPreviousStepTest {
+
+        @DisplayName("직전 사다리가 존재하는 경우, 사다리 발판을 생성하지 않는다.")
+        @Test
+        void previousStepExistsTest() {
+            assertThat(LadderStep.createConsideringPreviousStep(existsGenerator, LadderStep.EXISTS))
+                    .isEqualTo(LadderStep.NONE);
+            assertThat(LadderStep.createConsideringPreviousStep(noneGenerator, LadderStep.EXISTS))
+                    .isEqualTo(LadderStep.NONE);
```

Comment:
```suggestion
            assertAll(
                () -> assertThat(LadderStep.createConsideringPreviousStep(existsGenerator, LadderStep.EXISTS))
                    .isEqualTo(LadderStep.NONE),
                () -> assertThat(LadderStep.createConsideringPreviousStep(noneGenerator, LadderStep.EXISTS))
                    .isEqualTo(LadderStep.NONE)
            );
```

여러 assert를 검증하기 위한 assertAll이라는 기능을 제공하고 있습니다 :)